plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/OpenShiftConnector.java

```diff
@@ -208,31 +209,36 @@ public void removeContainer(final RemoveContainerParams params) throws IOExcepti
         boolean useForce = params.isForce();
         boolean removeVolumes = params.isRemoveVolumes();
 
-        IPod pod = getChePodByContainerId(containerId);
+        Pod pod = getChePodByContainerId(containerId);
 
-        String deploymentConfig = pod.getLabels().get("deploymentConfig");
-        String replicationController = pod.getLabels().get("deployment");
+        String deploymentConfig = pod.getMetadata().getLabels().get("deploymentConfig");
 
-        IDeploymentConfig dc = getCheDeploymentDescriptor(deploymentConfig);
-        IReplicationController rc = getCheReplicationController(replicationController);
-        IService svc = getCheServiceBySelector("deploymentConfig", deploymentConfig);
+        DeploymentConfig dc = getCheDeployment(deploymentConfig);
+        io.fabric8.kubernetes.api.model.Service svc = getCheServiceBySelector("deploymentConfig", deploymentConfig);
 
-        LOG.info("Removing OpenShift Service " + svc.getName());
-        openShiftClient.delete(svc);
-
-        LOG.info("Removing OpenShift DeploymentConfiguration " + dc.getName());
-        openShiftClient.delete(dc);
+        if (svc != null ) {
+            LOG.info("Removing OpenShift Service " + svc.getMetadata().getName());
+            openShiftClient.resource(svc).delete();
+        }
 
-        LOG.info("Removing OpenShift ReplicationController " + rc.getName());
-        openShiftClient.delete(rc);
+        if ( dc != null ) {
+            LOG.info("Removing OpenShift DeploymentConfiguration " + dc.getMetadata().getName());
+            openShiftClient.resource(dc).delete();
+        }
 
-        LOG.info("Removing OpenShift Pod " + pod.getName());
-        openShiftClient.delete(pod);
+        if ( pod != null ) {
+            LOG.info("Removing OpenShift Pod " + pod.getMetadata().getName());
+            openShiftClient.resource(pod).delete();
```

Comment:
I think this may be causing an issue where OpenShift prevents full deletion of the pod. After I stop a workspace, OpenShift immediately recreates the pod, which is then stuck in ImagePullBackOff

```
> oc get pods
NAME                                       READY     STATUS             RESTARTS   AGE
che-host-1-oas8k                           1/1       Running            0          1h
che-ws-946kcw4tm6djd9k9-1058218227-rkm5w   0/1       ImagePullBackOff   0          5m
che-ws-y714ql3djvc4q1dy-1150559083-dyo0h   0/1       ImagePullBackOff   0          6m
router-1-wotpb                             2/2       Running            10         20d
```

These pods have replica sets that are not removed, and so they are recreated. Since a pod with the name exists, the corresponding workspace cannot be restarted.

```  
> oc get replicasets
NAME                                 DESIRED   CURRENT   AGE
che-ws-946kcw4tm6djd9k9-1058218227   1         1         3m
che-ws-y714ql3djvc4q1dy-1150559083   1         1         1h

>oc get deployments
NAME                      DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
che-ws-946kcw4tm6djd9k9   1         1         1            0           21m
che-ws-y714ql3djvc4q1dy   1         1         1            0           1h
```